### ISSUE TITLE
Sealed secrets chart  v2.6.3 throwing errors on null spec.template.data field

### ISSUE THREAD

**Issue by brokenjacobs** [bitnami-labs/sealed-secrets#955]
Seeing multiple errors trying to dry-run in sealedsecrets manifests from my repo with flux that were working last week. All of them show a variation on this error with just names being different:

```
SealedSecret/cert-manager/**** dry-run failed, reason: Invalid, error: SealedSecret.bitnami.com "*****" is invalid: spec.template.data: Invalid value: "null": spec.template.data in body must be of type object: "null"
```

The relevant section of my manifests (generated with kubeseal) look like this:
```yaml
kind: SealedSecret
apiVersion: bitnami.com/v1alpha1
metadata:
  name: ******
  namespace: *****
spec:
  template:
    metadata:
      name: ****
      namespace: ****
      creationTimestamp: null
    data: null
```
This was deployed with helm chart v2.6.3. On k3s v1.24.4+k3s1

**Comment by brokenjacobs** [comment]
Looks like the 2.6.3 chart includes a new CRD spec with OpenAPI definitions that break with kubeseal generated manifests.

```
              template:
                description: Template defines the structure of the Secret that will
                  be created from this sealed secret.
                properties:
                  data:
                    additionalProperties:
                      type: string
                    description: Keys that should be templated using decrypted data
                    type: object
                  metadata:
                    description: 'Standard object''s metadata. More info: https://git.k8s.io/community/contributors/devel/api-conventions.md#metadata'
                    type: object
                  type:
                    description: Used to facilitate programmatic handling of secret
                      data.
                    type: string
```
I believe just deleting the keys from the manifest will fix the error being reported as a null is not of type 'string', and it's not a required field. But kubeseal producing manifests with these keys is problematic...

**Comment by cdenneen** [comment]
I have `Secret` with `data:` which yields a `SealedSecret` with `encryptedData:` but still generates a `spec.template.data: null` field causing major issue for reconciling.

**Comment by cdenneen** [comment]
For anyone with this roadblock I uninstalled `sealed-secrets` release from helm, manually delete the CRD, locked install version to `2.6.2` and now my reconciliation is working properly. (FYI This was breaking my Flux kustomize-controller, so now that is fixed.)

**Comment by pelumi1163** [comment]
> For anyone with this roadblock I uninstalled `sealed-secrets` release from helm, manually delete the CRD, locked install version to `2.6.2` and now my reconciliation is working properly. (FYI This was breaking my Flux kustomize-controller, so now that is fixed.)

This was also what I did.
Locked my version to 2.6.2

**Comment by brokenjacobs** [comment]
I had to apply a patch to my helmrelease to get things going again on 47 clusters. Fortunately system update controller was still taking updates via flux.

**Comment by JVMartin** [comment]
This broke all our flux-controlled clusters as well.

I worked around this by applying this patch to remove `data: null` every SealedSecret on our clusters:
```
   template:
-    data: null
     metadata:
       annotations:
```

**Comment by CapKenR** [comment]
We had the same problem with an additional complaint about a null creationTimestamp. We also "fixed" the problem by rolling back to 2.6.2.

**Comment by robkooper** [comment]
I had to roll back as well to 2.6.2. In my case it was the null value that was inserted by kubeseal in template section:
```
apiVersion: bitnami.com/v1alpha1
kind: SealedSecret
metadata:
  creationTimestamp: null
  name: repo-XYZ
  namespace: argocd
spec:
  encryptedData:
     PAYLOAD
  template:
    data: null
    metadata:
      creationTimestamp: null
      labels:
        argocd.argoproj.io/secret-type: repository
      name: repo-XYZ
      namespace: argocd
```

I ended up with the secrets but no labels, which broke argocd since it could not find the secret for the repo.

**Comment by grglzrv** [comment]
the same errors, rollback to 2.6.2 as well.

**Comment by adawalli** [comment]
Just for those following, the steps would be 

- Delete Helm Chart of current Controller
- If using flux, suspend any dependent kustomizations
- Delete the CRD (Very important) `k  delete customresourcedefinitions.apiextensions.k8s.io sealedsecrets.bitnami.com`
- Reinstall Helm Chart pinned to `2.6.2` (or apply your kustomization through flux)
- If using flux, resume any dependent kustomizations - this will force reconciliation with the older (unbroken) CRD

**Comment by alemorcuq** [comment]
Thanks for reporting this! I've created #957, which on my side fixes the validation issues.

**Comment by alemorcuq** [comment]
We released chart version 2.6.4, could you confirm the validation issues are gone?

**Comment by brokenjacobs** [comment]
Works for me just tested. Thanks!
